--- Python-2.4.2/Doc/info/Makefile~	2003-07-16 06:02:58.000000000 +0200
+++ Python-2.4.2/Doc/info/Makefile	2006-03-26 13:37:34.000000000 +0200
@@ -34,10 +34,10 @@
 
 check-emacs-version:
 	@v="`$(EMACS) --version 2>&1 | egrep '^(GNU |X)Emacs [12]*'`"; \
-	if `echo "$$v" | grep '^GNU Emacs 21' >/dev/null 2>&1`; then \
+	if `echo "$$v" | grep '^GNU Emacs 2[1-9]' >/dev/null 2>&1`; then \
 	  echo "Using $(EMACS) to build the info docs"; \
 	else \
-	  echo "GNU Emacs 21 is required to build the info docs"; \
+	  echo "GNU Emacs 2[1-9] is required to build the info docs"; \
 	  echo "Found $$v"; \
 	  false; \
 	fi
--- Python-2.4.2/Doc/tools/mkinfo~	2003-09-29 19:26:08.000000000 +0200
+++ Python-2.4.2/Doc/tools/mkinfo	2006-03-26 13:52:20.000000000 +0200
@@ -61,5 +61,5 @@
 
 # generate the .info files:
 
-run $MAKEINFO --footnote-style end --fill-column 72 \
+run $MAKEINFO --footnote-style end --fill-column 72 --no-split \
 	      --paragraph-indent 0 --output=$INFONAME $TEXINAME
