--- Python-2.2/configure.in~	Mon Feb  4 06:57:41 2002
+++ Python-2.2/configure.in	Mon Feb  4 06:57:41 2002
@@ -390,10 +390,10 @@
 		# debug builds.
 		OPT="-g -Wall -Wstrict-prototypes"
 	    else
-		OPT="-g -O3 -Wall -Wstrict-prototypes"
+		OPT="$CFLAGS -Wall -Wstrict-prototypes"
 	    fi;;
 	*)
-	    OPT="-O3 -Wall -Wstrict-prototypes";;
+	    OPT="$CFLAGS -Wall -Wstrict-prototypes";;
 	esac
 	;;
     *)
