diff -Nur Python-2.2.2.orig/Modules/readline.c Python-2.2.2/Modules/readline.c
--- Python-2.2.2.orig/Modules/readline.c	Sun Oct  6 07:43:47 2002
+++ Python-2.2.2/Modules/readline.c	Thu Oct 17 12:10:51 2002
@@ -450,7 +450,7 @@
 /* C function to call the Python completer. */
 
 static char *
-on_completion(char *text, int state)
+on_completion(const char *text, int state)
 {
 	char *result = NULL;
 	if (completer != NULL) {
@@ -495,7 +495,7 @@
 	Py_XDECREF(endidx);
 	begidx = PyInt_FromLong((long) start);
 	endidx = PyInt_FromLong((long) end);
-	return completion_matches(text, *on_completion);
+	return rl_completion_matches(text, *on_completion);
 }
 
 /* Helper to initialize GNU readline properly. */
