--- Python-2.5b2/configure.in.orig	2006-07-06 11:13:35.000000000 +0100
+++ Python-2.5b2/configure.in	2006-07-12 16:30:30.000000000 +0100
@@ -753,13 +753,13 @@
 	    if test "$Py_DEBUG" = 'true' ; then
 		# Optimization messes up debuggers, so turn it off for
 		# debug builds.
-		OPT="-g -Wall $STRICT_PROTO"
+		OPT="-g $CPPFLAGS $CFLAGS -Wall $STRICT_PROTO"
 	    else
-		OPT="-g -O3 -Wall $STRICT_PROTO"
+		OPT="-g $CPPFLAGS $CFLAGS -Wall $STRICT_PROTO"
 	    fi
 	    ;;
 	*)
-	    OPT="-O3 -Wall $STRICT_PROTO"
+	    OPT="$CPPFLAGS $CFLAGS -Wall $STRICT_PROTO"
 	    ;;
 	esac
 	case $ac_sys_system in
