diff -Nur Python-2.1.orig/Makefile.pre.in Python-2.1/Makefile.pre.in
--- Python-2.1.orig/Makefile.pre.in	Sat Apr 14 19:57:07 2001
+++ Python-2.1/Makefile.pre.in	Thu May 24 17:57:09 2001
@@ -25,6 +25,7 @@
 
 # === Variables set by configure
 VERSION=	@VERSION@
+SOVERSION=	@SOVERSION@
 srcdir=		@srcdir@
 VPATH=		@srcdir@
 
@@ -64,7 +65,7 @@
 # Extra C flags added for building the interpreter object files.
 CFLAGSFORSHARED=@CFLAGSFORSHARED@
 # C flags used for building the interpreter object files
-PY_CFLAGS=	$(CFLAGS) $(CFLAGSFORSHARED)
+PY_CFLAGS=	$(CFLAGS)
 
 
 # Machine-dependent subdirectories
@@ -96,6 +97,7 @@
 LDSHARED=	@LDSHARED@
 BLDSHARED=	@BLDSHARED@
 DESTSHARED=	$(BINLIBDEST)/lib-dynload
+RUNSHARED=	LD_PRELOAD=./$(LDLIBRARY)
 
 # Executable suffix (.exe on Windows and Mac OS X)
 EXE=		@EXEEXT@
@@ -268,7 +270,7 @@
 # Rules
 
 # Default target
-all:		$(PYTHON) oldsharedmods sharedmods 
+all:		$(PYTHON) $(LIBRARY) oldsharedmods sharedmods 
 
 # Build the interpreter
 $(PYTHON):	Modules/$(MAINOBJ) $(LDLIBRARY)
@@ -277,12 +279,12 @@
 			$(LDLIBRARY) $(LIBS) $(MODLIBS) $(SYSLIBS) $(LDLAST)
 
 platform: $(PYTHON)
-	./$(PYTHON) -c 'import sys ; from distutils.util import get_platform ; print get_platform()+"-"+sys.version[0:3]' >platform
+	$(RUNSHARED) ./$(PYTHON) -c 'import sys ; from distutils.util import get_platform ; print get_platform()+"-"+sys.version[0:3]' >platform
 
 
 # Build the shared modules
 sharedmods: $(PYTHON)
-	PYTHONPATH= ./$(PYTHON) $(srcdir)/setup.py build
+	PYTHONPATH= $(RUNSHARED) ./$(PYTHON) $(srcdir)/setup.py build
 
 # buildno should really depend on something like LIBRARY_SRC
 buildno: $(PARSER_OBJS) \
@@ -309,6 +311,19 @@
 	$(AR) cr $@ $(MODOBJS)
 	$(RANLIB) $@
 
+# WARNING: The following is only tested for Debian GNU/Linux.
+# This rule is silly and needs to be cleaned up ASAP!
+#      Gregor Hoffleit <flight@debian.org>
+#
+# This rule is here for Linux shared libs
+# The PIC code will be built additionally to the static lib!
+libpython$(VERSION).so.$(SOVERSION): $(LIBRARY_OBJS:%.o=%.pic_o)
+	case `uname -s | tr -d '/ ' | tr '[A-Z]' '[a-z]'` in \
+	*linux*) \
+	    $(CC) -shared -Wl,-soname,$@ -o $@ $(LIBRARY_OBJS:%.o=%.pic_o) $(LIBS) $(LIBM) \
+	    ;; \
+	esac
+
 # This rule is only here for DG/UX, UnixWare, and BeOS!!!
 libpython$(VERSION).so:	$(LIBRARY)
 	case `uname -s | tr -d '/ ' | tr '[A-Z]' '[a-z]'` in \
@@ -370,6 +385,9 @@
 Modules/getbuildinfo.o: $(srcdir)/Modules/getbuildinfo.c buildno
 	$(CC) -c $(PY_CFLAGS) -DBUILD=`cat buildno` -o $@ $(srcdir)/Modules/getbuildinfo.c
 
+Modules/getbuildinfo.pic_o: $(srcdir)/Modules/getbuildinfo.c buildno
+	$(CC) -c $(PY_CFLAGS) $(CCSHARED) -DBUILD=`cat buildno` -o $@ $(srcdir)/Modules/getbuildinfo.c
+
 Modules/getpath.o: $(srcdir)/Modules/getpath.c Makefile
 	$(CC) -c $(PY_CFLAGS) -DPYTHONPATH='"$(PYTHONPATH)"' \
 		-DPREFIX='"$(prefix)"' \
@@ -378,9 +396,20 @@
 		-DVPATH='"$(VPATH)"' \
 		-o $@ $(srcdir)/Modules/getpath.c
 
+Modules/getpath.pic_o: $(srcdir)/Modules/getpath.c Makefile
+	$(CC) -c $(PY_CFLAGS) $(CCSHARED) -DPYTHONPATH='"$(PYTHONPATH)"' \
+		-DPREFIX='"$(prefix)"' \
+		-DEXEC_PREFIX='"$(exec_prefix)"' \
+		-DVERSION='"$(VERSION)"' \
+		-DVPATH='"$(VPATH)"' \
+		-o $@ $(srcdir)/Modules/getpath.c
+
 Modules/ccpython.o: $(srcdir)/Modules/ccpython.cc
 	$(CXX) -c $(PY_CFLAGS) -o $@ $(srcdir)/Modules/ccpython.cc
 
+Modules/ccpython.pic_o: $(srcdir)/Modules/ccpython.cc
+	$(CXX) -c $(PY_CFLAGS) $(CCSHARED) -o $@ $(srcdir)/Modules/ccpython.cc
+
 
 $(GRAMMAR_H) $(GRAMMAR_C): $(PGEN) $(GRAMMAR_INPUT)
 		-$(PGEN) $(GRAMMAR_INPUT) $(GRAMMAR_H) $(GRAMMAR_C)
@@ -400,9 +429,15 @@
 Python/getplatform.o: $(srcdir)/Python/getplatform.c
 		$(CC) -c $(CFLAGS) -DPLATFORM='"$(MACHDEP)"' -o $@ $(srcdir)/Python/getplatform.c
 
+Python/getplatform.pic_o: $(srcdir)/Python/getplatform.c
+		$(CC) -c $(CFLAGS) $(CCSHARED) -DPLATFORM='"$(MACHDEP)"' -o $@ $(srcdir)/Python/getplatform.c
+
 Python/importdl.o: $(srcdir)/Python/importdl.c
 		$(CC) -c $(CFLAGS) -I$(DLINCLDIR) -o $@ $(srcdir)/Python/importdl.c
 
+Python/importdl.pic_o: $(srcdir)/Python/importdl.c
+		$(CC) -c $(CFLAGS) $(CCSHARED) -I$(DLINCLDIR) -o $@ $(srcdir)/Python/importdl.c
+
 Objects/object.o: $(srcdir)/Objects/object.c $(srcdir)/Objects/obmalloc.c
 
 Objects/unicodectype.o:	$(srcdir)/Objects/unicodectype.c \
@@ -462,7 +497,7 @@
 # Test the interpreter (twice, once without .pyc files, once with)
 TESTOPTS=	-l
 TESTPROG=	$(srcdir)/Lib/test/regrtest.py
-TESTPYTHON=	./$(PYTHON) -tt
+TESTPYTHON=	$(RUNSHARED) ./$(PYTHON) -tt
 test:		all platform
 		-find $(srcdir)/Lib -name '*.py[co]' -print | xargs rm -f
 		-PYTHONPATH= $(TESTPYTHON) $(TESTPROG) $(TESTOPTS)
@@ -532,6 +567,12 @@
 		fi; \
 	done
 	$(INSTALL_PROGRAM) $(PYTHON) $(BINDIR)/python$(VERSION)$(EXE)
+	if test -f libpython$(VERSION).so.$(SOVERSION); then \
+		mkdir $(LIBDIR); \
+		$(INSTALL_DATA) libpython$(VERSION).so.$(SOVERSION) $(LIBDIR); \
+		(cd $(LIBDIR) && $(LN) -sf libpython$(VERSION).so.$(SOVERSION) libpython$(VERSION).so); \
+	else	true; \
+	fi
 	if test -f libpython$(VERSION).so; then \
 		$(INSTALL_DATA) libpython$(VERSION).so $(LIBDIR); \
 	else	true; \
@@ -615,10 +656,10 @@
 		done; \
 	done
 	$(INSTALL_DATA) $(srcdir)/LICENSE $(LIBDEST)/LICENSE.txt
-	PYTHONPATH=$(LIBDEST) \
-		./$(PYTHON) -tt $(LIBDEST)/compileall.py $(LIBDEST)
-	PYTHONPATH=$(LIBDEST) \
-		./$(PYTHON) -O $(LIBDEST)/compileall.py $(LIBDEST)
+##	PYTHONPATH=$(LIBDEST) \
+##		$(RUNSHARED) ./$(PYTHON) -tt $(LIBDEST)/compileall.py $(LIBDEST)
+##	PYTHONPATH=$(LIBDEST) \
+##		$(RUNSHARED) ./$(PYTHON) -O $(LIBDEST)/compileall.py $(LIBDEST)
 
 # Create the PLATDIR source directory, if one wasn't distributed..
 $(srcdir)/Lib/$(PLATDIR):
@@ -659,9 +700,9 @@
 		else	true; \
 		fi; \
 	done
-	@if test -d $(LDLIBRARY); then :; else \
-		$(INSTALL_DATA) $(LDLIBRARY) $(LIBPL)/$(LDLIBRARY) ; \
-		$(RANLIB) $(LIBPL)/$(LDLIBRARY) ; \
+	@if test -d $(LIBRARY); then :; else \
+		$(INSTALL_DATA) $(LIBRARY) $(LIBPL)/$(LIBRARY) ; \
+		$(RANLIB) $(LIBPL)/$(LIBRARY) ; \
 	fi
 	$(INSTALL_DATA) Modules/config.c $(LIBPL)/config.c
 	$(INSTALL_DATA) Modules/$(MAINOBJ) $(LIBPL)/$(MAINOBJ)
@@ -703,7 +744,7 @@
 # Install the dynamically loadable modules
 # This goes into $(exec_prefix)
 sharedinstall:
-	PYTHONPATH= ./$(PYTHON) $(srcdir)/setup.py install \
+	PYTHONPATH= $(RUNSHARED) ./$(PYTHON) $(srcdir)/setup.py install \
 		--install-platlib=$(DESTSHARED)
 
 # Build the toplevel Makefile
@@ -721,6 +762,11 @@
 .c.o:
 	$(CC) -c $(PY_CFLAGS) -o $@ $<
 
+.SUFFIXES: .pic_o
+
+.c.pic_o:
+	$(CC) -c $(PY_CFLAGS) $(CCSHARED) -o $@ $<
+
 # Rerun configure with the same options as it was run last time,
 # provided the config.status script exists
 recheck:
@@ -751,6 +797,7 @@
 
 clean:
 	find . -name '*.o' -exec rm -f {} ';'
+	find . -name '*.pic_o' -exec rm -f {} ';'
 	find . -name '*.s[ol]' -exec rm -f {} ';'
 	find $(srcdir) -name '*.py[co]' -exec rm -f {} ';'
 
diff -Nur Python-2.1.orig/configure.in Python-2.1/configure.in
--- Python-2.1.orig/configure.in	Wed Apr 11 22:56:19 2001
+++ Python-2.1/configure.in	Thu May 24 17:57:09 2001
@@ -8,6 +8,10 @@
 AC_SUBST(VERSION)
 VERSION=2.1
 
+# Set SOVERSION
+AC_SUBST(SOVERSION)
+SOVERSION=0.0
+
 # Arguments passed to configure.
 AC_SUBST(CONFIG_ARGS)
 CONFIG_ARGS="$ac_configure_args"
@@ -235,6 +239,7 @@
 # LINKCC is the command that links the python executable -- default is $(CC).
 # This is altered for AIX in order to build the export list before 
 # linking.
+# For Linux, we now build a shared library, too.
 AC_SUBST(LINKCC)
 AC_MSG_CHECKING(LINKCC)
 if test -z "$LINKCC"
@@ -246,6 +251,9 @@
 	   LINKCC="LD_RUN_PATH=$libdir \$(PURIFY) \$(CC)";;
 	Monterey64*)
 	   LINKCC="\$(PURIFY) \$(CC) -L/usr/lib/ia64l64";;
+	Linux*)
+	   LDLIBRARY='libpython$(VERSION).so.$(SOVERSION)'
+	   LINKCC="\$(CC)";;
 	*) LINKCC="\$(PURIFY) \$(CC)";;
 	esac
 fi
@@ -596,7 +604,7 @@
 	if test "$with_next_framework" ; then
 		    LDSHARED="$LDSHARED \$(LDLIBRARY)"
 		fi ;;
-	Linux*) LDSHARED="gcc -shared";;
+	Linux*) LDSHARED="gcc -shared -lc";;
 	dgux*) LDSHARED="ld -G";;
 	BSD/OS*/4*) LDSHARED="gcc -shared";;
 	OpenBSD*) LDSHARED="ld -Bshareable";;
