diff -urNp -x '*.orig' Python-2.7.18.org/Makefile.pre.in Python-2.7.18/Makefile.pre.in
--- Python-2.7.18.org/Makefile.pre.in	2020-04-19 23:13:39.000000000 +0200
+++ Python-2.7.18/Makefile.pre.in	2021-09-26 12:34:28.007790961 +0200
@@ -1030,14 +1030,12 @@ altmaninstall:
 		fi; \
 	done
 	$(INSTALL_DATA) $(srcdir)/Misc/python.man \
-		$(DESTDIR)$(MANDIR)/man1/python$(VERSION).1
+		$(DESTDIR)$(MANDIR)/man1/python.1
 
 # Install the unversioned manual pages
 maninstall:	altmaninstall
 	-rm -f $(DESTDIR)$(MANDIR)/man1/python2.1
-	(cd $(DESTDIR)$(MANDIR)/man1; $(LN) -s python$(VERSION).1 python2.1)
-	-rm -f $(DESTDIR)$(MANDIR)/man1/python.1
-	(cd $(DESTDIR)$(MANDIR)/man1; $(LN) -s python2.1 python.1)
+	(cd $(DESTDIR)$(MANDIR)/man1; $(LN) -s python.1 python2.1)
 
 # Install the library
 PLATDIR=	@PLATDIR@
@@ -1180,7 +1178,7 @@ $(srcdir)/Lib/$(PLATDIR):
 python-config: $(srcdir)/Misc/python-config.in
 	# Substitution happens here, as the completely-expanded BINDIR
 	# is not available in configure
-	sed -e "s,@EXENAME@,$(BINDIR)/python$(VERSION)$(EXE)," < $(srcdir)/Misc/python-config.in >python-config
+	sed -e "s,@EXENAME@,$(BINDIR)/python$(EXE)," < $(srcdir)/Misc/python-config.in >python-config
 
 # Install the include files
 INCLDIRSTOMAKE=$(INCLUDEDIR) $(CONFINCLUDEDIR) $(INCLUDEPY) $(CONFINCLUDEPY)
diff -urNp -x '*.orig' Python-2.7.18.org/configure.ac Python-2.7.18/configure.ac
--- Python-2.7.18.org/configure.ac	2020-04-19 23:13:39.000000000 +0200
+++ Python-2.7.18/configure.ac	2021-09-26 12:34:28.007790961 +0200
@@ -1090,13 +1090,13 @@ then
 	    if test "$Py_DEBUG" = 'true' ; then
 		# Optimization messes up debuggers, so turn it off for
 		# debug builds.
-		OPT="-g -O0 -Wall $STRICT_PROTO"
+		OPT="-g $CPPFLAGS $CFLAGS -Wall $STRICT_PROTO"
 	    else
-		OPT="-g $WRAP -O3 -Wall $STRICT_PROTO"
+		OPT="-g $WRAP $CPPFLAGS $CFLAGS -Wall $STRICT_PROTO"
 	    fi
 	    ;;
 	*)
-	    OPT="-O3 -Wall $STRICT_PROTO"
+	    OPT="$CPPFLAGS $CFLAGS -Wall $STRICT_PROTO"
 	    ;;
 	esac
 	case $ac_sys_system in
