diff -Nur Python-2.3a2.orig/configure.in Python-2.3a2/configure.in
--- Python-2.3a2.orig/configure.in	Wed Feb 19 16:25:07 2003
+++ Python-2.3a2/configure.in	Wed Apr  9 15:05:25 2003
@@ -568,11 +568,11 @@
 		# debug builds.
 		OPT="-g -Wall -Wstrict-prototypes"
 	    else
-		OPT="-g -O3 -Wall -Wstrict-prototypes"
+		OPT="-g $CFLAGS -Wstrict-prototypes"
 	    fi
 	    ;;
 	*)
-	    OPT="-O3 -Wall -Wstrict-prototypes"
+	    OPT="$CFLAGS -Wstrict-prototypes"
 	    ;;
 	esac
 	case $ac_sys_system in
