Index: Python/ast.c
===================================================================
--- Python/ast.c	(wersja 52621)
+++ Python/ast.c	(wersja 52622)
@@ -538,6 +538,7 @@
 	   || TYPE(n) == listmaker
 	   || TYPE(n) == testlist_gexp
 	   || TYPE(n) == testlist_safe
+	   || TYPE(n) == testlist1
 	   );
 
     seq = asdl_seq_new((NCH(n) + 1) / 2, c->c_arena);
Index: Lib/test/test_grammar.py
===================================================================
--- Lib/test/test_grammar.py	(wersja 52621)
+++ Lib/test/test_grammar.py	(wersja 52622)
@@ -709,6 +709,7 @@
 
 x = `x`
 x = `1 or 2 or 3`
+x = `1,2`
 x = x
 x = 'x'
 x = 123
