diff -dur Python-2.3.4.orig/Lib/distutils/command/install.py Python-2.3.4/Lib/distutils/command/install.py
--- Python-2.3.4.orig/Lib/distutils/command/install.py	2002-11-19 14:12:28.000000000 +0100
+++ Python-2.3.4/Lib/distutils/command/install.py	2004-09-10 08:41:12.836021548 +0200
@@ -38,14 +38,14 @@
 
 INSTALL_SCHEMES = {
     'unix_prefix': {
-        'purelib': '$base/lib/python$py_version_short/site-packages',
+        'purelib': '$base/share/python$py_version_short/site-packages',
         'platlib': '$platbase/lib/python$py_version_short/site-packages',
         'headers': '$base/include/python$py_version_short/$dist_name',
         'scripts': '$base/bin',
         'data'   : '$base',
         },
     'unix_home': {
-        'purelib': '$base/lib/python',
+        'purelib': '$base/share/python',
         'platlib': '$base/lib/python',
         'headers': '$base/include/python/$dist_name',
         'scripts': '$base/bin',
diff -dur Python-2.3.4.orig/Lib/distutils/sysconfig.py Python-2.3.4/Lib/distutils/sysconfig.py
--- Python-2.3.4.orig/Lib/distutils/sysconfig.py	2003-02-10 15:02:33.000000000 +0100
+++ Python-2.3.4/Lib/distutils/sysconfig.py	2004-09-10 08:44:08.196435115 +0200
@@ -99,8 +99,12 @@
         prefix = plat_specific and EXEC_PREFIX or PREFIX
 
     if os.name == "posix":
-        libpython = os.path.join(prefix,
+        if plat_specific: 
+            libpython = os.path.join(prefix,
                                  "lib", "python" + get_python_version())
+        else:
+            libpython = os.path.join(prefix,
+                                 "share", "python" + get_python_version())
         if standard_lib:
             return libpython
         else:
diff -dur Python-2.3.4.orig/Lib/site.py Python-2.3.4/Lib/site.py
--- Python-2.3.4.orig/Lib/site.py	2004-09-10 08:26:38.221827272 +0200
+++ Python-2.3.4/Lib/site.py	2004-09-10 08:41:12.837021494 +0200
@@ -169,7 +169,12 @@
                                      "lib",
                                      "python" + sys.version[:3],
                                      "site-packages"),
-                        os.path.join(prefix, "lib", "site-python")]
+                        os.path.join(prefix, "lib", "site-python"),
+                        os.path.join(prefix,
+                                     "share",
+                                     "python" + sys.version[:3],
+                                     "site-packages"),
+                        os.path.join(prefix, "share", "site-python")]
         else:
             sitedirs = [prefix, os.path.join(prefix, "lib", "site-packages")]
         if sys.platform == 'darwin':
diff -dur Python-2.3.4.orig/Makefile.pre.in Python-2.3.4/Makefile.pre.in
--- Python-2.3.4.orig/Makefile.pre.in	2003-11-18 20:54:00.000000000 +0100
+++ Python-2.3.4/Makefile.pre.in	2004-09-10 08:41:12.838021439 +0200
@@ -83,7 +83,7 @@
 MANDIR=		@mandir@
 INCLUDEDIR=	@includedir@
 CONFINCLUDEDIR=	$(exec_prefix)/include
-SCRIPTDIR=	$(prefix)/lib
+SCRIPTDIR=	$(prefix)/share
 
 # Detailed destination directories
 BINLIBDEST=	$(LIBDIR)/python$(VERSION)
diff -dur Python-2.3.4.orig/Modules/getpath.c Python-2.3.4/Modules/getpath.c
--- Python-2.3.4.orig/Modules/getpath.c	2002-12-31 13:45:12.000000000 +0100
+++ Python-2.3.4/Modules/getpath.c	2004-09-10 08:41:12.840021330 +0200
@@ -112,7 +112,7 @@
 #endif
 
 #ifndef PYTHONPATH
-#define PYTHONPATH PREFIX "/lib/python" VERSION ":" \
+#define PYTHONPATH PREFIX "/share/python" VERSION ":" \
               EXEC_PREFIX "/lib/python" VERSION "/lib-dynload"
 #endif
 
@@ -125,6 +125,7 @@
 static char progpath[MAXPATHLEN+1];
 static char *module_search_path = NULL;
 static char lib_python[] = "lib/python" VERSION;
+static char share_python[] = "share/python" VERSION;
 
 static void
 reduce(char *dir)
@@ -256,7 +257,7 @@
         delim = strchr(prefix, DELIM);
         if (delim)
             *delim = '\0';
-        joinpath(prefix, lib_python);
+        joinpath(prefix, share_python);
         joinpath(prefix, LANDMARK);
         return 1;
     }
@@ -279,7 +280,7 @@
     copy_absolute(prefix, argv0_path);
     do {
         n = strlen(prefix);
-        joinpath(prefix, lib_python);
+        joinpath(prefix, share_python);
         joinpath(prefix, LANDMARK);
         if (ismodule(prefix))
             return 1;
@@ -289,7 +290,7 @@
 
     /* Look at configure's PREFIX */
     strncpy(prefix, PREFIX, MAXPATHLEN);
-    joinpath(prefix, lib_python);
+    joinpath(prefix, share_python);
     joinpath(prefix, LANDMARK);
     if (ismodule(prefix))
         return 1;
@@ -479,7 +480,7 @@
             fprintf(stderr,
                 "Could not find platform independent libraries <prefix>\n");
         strncpy(prefix, PREFIX, MAXPATHLEN);
-        joinpath(prefix, lib_python);
+        joinpath(prefix, share_python);
     }
     else
         reduce(prefix);
