--- Python-2.3.3/Lib/pydoc.py.orig	2004-01-07 12:20:29.000000000 +0100
+++ Python-2.3.3/Lib/pydoc.py	2004-01-07 12:23:05.000000000 +0100
@@ -1527,12 +1527,7 @@
         homedir = os.environ.get('PYTHONHOME')
         for dir in [os.environ.get('PYTHONDOCS'),
                     homedir and os.path.join(homedir, 'doc'),
-                    os.path.join(execdir, 'doc'),
-                    '/usr/doc/python-docs-' + split(sys.version)[0],
-                    '/usr/doc/python-' + split(sys.version)[0],
-                    '/usr/doc/python-docs-' + sys.version[:3],
-                    '/usr/doc/python-' + sys.version[:3],
-                    os.path.join(sys.prefix, 'Resources/English.lproj/Documentation')]:
+                    '/usr/share/doc/python-doc-' + split(sys.version)[0]]:
             if dir and os.path.isdir(os.path.join(dir, 'lib')):
                 self.docdir = dir
 
