--- Python-2.7.4/Makefile.pre.in~	2013-04-23 18:32:25.000000000 +0200
+++ Python-2.7.4/Makefile.pre.in	2013-04-23 18:34:24.626857195 +0200
@@ -450,7 +450,7 @@
 sharedmods: $(BUILDPYTHON)
 	@case "$$MAKEFLAGS" in \
 	    *\ -s*|s*) quiet="-q";; \
-	    *) quiet="";; \
+	    *) quiet="-vvv";; \
 	esac; \
 	$(RUNSHARED) CC='$(CC)' LDSHARED='$(BLDSHARED)' OPT='$(OPT)' \
 		$(PYTHON_FOR_BUILD) $(srcdir)/setup.py $$quiet build
