diff -dur Python-2.3.3.orig/Lib/distutils/command/install.py Python-2.3.3/Lib/distutils/command/install.py
--- Python-2.3.3.orig/Lib/distutils/command/install.py	2004-01-15 18:20:10.091035345 +0100
+++ Python-2.3.3/Lib/distutils/command/install.py	2004-01-15 18:20:27.037955276 +0100
@@ -39,14 +39,14 @@
 INSTALL_SCHEMES = {
     'unix_prefix': {
         'purelib': '$base/share/python$py_version_short/site-packages',
-        'platlib': '$platbase/lib/python$py_version_short/site-packages',
+        'platlib': '$platbase/lib64/python$py_version_short/site-packages',
         'headers': '$base/include/python$py_version_short/$dist_name',
         'scripts': '$base/bin',
         'data'   : '$base',
         },
     'unix_home': {
         'purelib': '$base/share/python',
-        'platlib': '$base/lib/python',
+        'platlib': '$base/lib64/python',
         'headers': '$base/include/python/$dist_name',
         'scripts': '$base/bin',
         'data'   : '$base',
diff -dur Python-2.3.3.orig/Lib/distutils/sysconfig.py Python-2.3.3/Lib/distutils/sysconfig.py
--- Python-2.3.3.orig/Lib/distutils/sysconfig.py	2003-02-10 15:02:33.000000000 +0100
+++ Python-2.3.3/Lib/distutils/sysconfig.py	2004-01-15 18:20:27.038955035 +0100
@@ -100,7 +100,7 @@
 
     if os.name == "posix":
         libpython = os.path.join(prefix,
-                                 "lib", "python" + get_python_version())
+                                 "lib64", "python" + get_python_version())
         if standard_lib:
             return libpython
         else:
diff -dur Python-2.3.3.orig/Lib/site.py Python-2.3.3/Lib/site.py
--- Python-2.3.3.orig/Lib/site.py	2004-01-15 18:20:10.091035345 +0100
+++ Python-2.3.3/Lib/site.py	2004-01-15 18:20:27.038955035 +0100
@@ -167,10 +167,10 @@
             sitedirs = [os.path.join(prefix, "Lib", "site-packages")]
         elif os.sep == '/':
             sitedirs = [os.path.join(prefix,
-                                     "lib",
+                                     "lib64",
                                      "python" + sys.version[:3],
                                      "site-packages"),
-                        os.path.join(prefix, "lib", "site-python"),
+                        os.path.join(prefix, "lib64", "site-python"),
 	    		os.path.join(prefix,
                                      "share",
                                      "python" + sys.version[:3],
diff -dur Python-2.3.3.orig/Makefile.pre.in Python-2.3.3/Makefile.pre.in
--- Python-2.3.3.orig/Makefile.pre.in	2004-01-15 18:20:10.092035105 +0100
+++ Python-2.3.3/Makefile.pre.in	2004-01-15 18:20:54.525341800 +0100
@@ -79,7 +79,7 @@
 
 # Expanded directories
 BINDIR=		$(exec_prefix)/bin
-LIBDIR=		$(exec_prefix)/lib
+LIBDIR=		$(exec_prefix)/lib64
 MANDIR=		@mandir@
 INCLUDEDIR=	@includedir@
 CONFINCLUDEDIR=	$(exec_prefix)/include
Only in Python-2.3.3: Makefile.pre.in.orig
Only in Python-2.3.3: Makefile.pre.in.rej
Only in Python-2.3.3: Makefile.pre.in~
diff -dur Python-2.3.3.orig/Modules/getpath.c Python-2.3.3/Modules/getpath.c
--- Python-2.3.3.orig/Modules/getpath.c	2002-12-31 13:45:12.000000000 +0100
+++ Python-2.3.3/Modules/getpath.c	2004-01-15 18:20:27.040954554 +0100
@@ -112,8 +112,8 @@
 #endif
 
 #ifndef PYTHONPATH
-#define PYTHONPATH PREFIX "/share/python" VERSION ":" \
-              EXEC_PREFIX "/lib/python" VERSION "/lib-dynload"
+#define PYTHONPATH PREFIX "/share/python" VERSION ":" \
+              EXEC_PREFIX "/lib64/python" VERSION "/lib-dynload"
 #endif
 
 #ifndef LANDMARK
@@ -124,7 +124,7 @@
 static char exec_prefix[MAXPATHLEN+1];
 static char progpath[MAXPATHLEN+1];
 static char *module_search_path = NULL;
-static char lib_python[] = "lib/python" VERSION;
+static char lib_python[] = "lib64/python" VERSION;
 
 static void
 reduce(char *dir)
@@ -492,7 +492,7 @@
     }
     else
         strncpy(zip_path, PREFIX, MAXPATHLEN);
-    joinpath(zip_path, "lib/python00.zip");
+    joinpath(zip_path, "lib64/python00.zip");
     bufsz = strlen(zip_path);	/* Replace "00" with version */
     zip_path[bufsz - 6] = VERSION[0];
     zip_path[bufsz - 5] = VERSION[2];
Only in Python-2.3.3.orig: Python-Docs-2.3.3
diff -dur Python-2.3.3.orig/setup.py Python-2.3.3/setup.py
--- Python-2.3.3.orig/setup.py	2004-01-15 18:20:10.086036549 +0100
+++ Python-2.3.3/setup.py	2004-01-15 18:20:27.040954554 +0100
@@ -264,7 +264,7 @@
         # lib_dirs and inc_dirs are used to search for files;
         # if a file is found in one of those directories, it can
         # be assumed that no additional -I,-L directives are needed.
-        lib_dirs = self.compiler.library_dirs + ['/lib', '/usr/lib']
+        lib_dirs = self.compiler.library_dirs + ['/lib64', '/usr/lib64']
         inc_dirs = self.compiler.include_dirs + ['/usr/include']
         exts = []
 
@@ -966,7 +966,7 @@
             added_lib_dirs.append('/usr/openwin/lib')
         elif os.path.exists('/usr/X11R6/include'):
             include_dirs.append('/usr/X11R6/include')
-            added_lib_dirs.append('/usr/X11R6/lib')
+            added_lib_dirs.append('/usr/X11R6/lib64')
         elif os.path.exists('/usr/X11R5/include'):
             include_dirs.append('/usr/X11R5/include')
             added_lib_dirs.append('/usr/X11R5/lib')
