--- Python-2.4.2/Lib/distutils/command/install.py~	2006-01-25 22:39:11.000000000 +0200
+++ Python-2.4.2/Lib/distutils/command/install.py	2006-01-25 22:42:30.000000000 +0200
@@ -40,14 +40,14 @@
 
 INSTALL_SCHEMES = {
     'unix_prefix': {
-        'purelib': '$base/lib/python$py_version_short/site-packages',
+        'purelib': '$base/share/python$py_version_short/site-packages',
         'platlib': '$platbase/'+libname+'/python$py_version_short/site-packages',
         'headers': '$base/include/python$py_version_short/$dist_name',
         'scripts': '$base/bin',
         'data'   : '$base',
         },
     'unix_home': {
-        'purelib': '$base/lib/python',
+        'purelib': '$base/share/python',
         'platlib': '$base/'+libname+'/python',
         'headers': '$base/include/python/$dist_name',
         'scripts': '$base/bin',
--- Python-2.4.2/Lib/distutils/sysconfig.py~	2006-01-25 22:39:11.000000000 +0200
+++ Python-2.4.2/Lib/distutils/sysconfig.py	2006-01-25 22:46:07.000000000 +0200
@@ -99,12 +99,12 @@
         prefix = plat_specific and EXEC_PREFIX or PREFIX
 
     if os.name == "posix":
-        if plat_specific:
-            lib = sys.lib
+        if plat_specific: 
+            libpython = os.path.join(prefix,
+                                 sys.lib, "python" + get_python_version())
         else:
-            lib = 'lib'
-        libpython = os.path.join(prefix,
-                                 lib, "python" + get_python_version())
+            libpython = os.path.join(prefix,
+                                 "share", "python" + get_python_version())
         if standard_lib:
             return libpython
         else:
--- Python-2.4.2/Makefile.pre.in~	2006-01-25 22:39:11.000000000 +0200
+++ Python-2.4.2/Makefile.pre.in	2006-01-25 22:47:54.000000000 +0200
@@ -85,7 +85,7 @@
 MANDIR=		@mandir@
 INCLUDEDIR=	@includedir@
 CONFINCLUDEDIR=	$(exec_prefix)/include
-SCRIPTDIR=	$(prefix)/$(LIB)
+SCRIPTDIR=	$(prefix)/share
 
 # Detailed destination directories
 BINLIBDEST=	$(LIBDIR)/python$(VERSION)
--- Python-2.4.2/Modules/getpath.c~	2006-01-25 22:39:11.000000000 +0200
+++ Python-2.4.2/Modules/getpath.c	2006-01-25 22:56:52.000000000 +0200
@@ -121,7 +121,8 @@
 
 #ifndef PYTHONPATH
 #define PYTHONPATH PREFIX "/" LIB_PYTHON ":" \
-              EXEC_PREFIX "/" LIB_PYTHON "/lib-dynload"
+              EXEC_PREFIX "/" LIB_PYTHON "/lib-dynload:" \
+              PREFIX "/share/python" VERSION
 #endif
 
 #ifndef LANDMARK
@@ -133,6 +133,7 @@
 static char progpath[MAXPATHLEN+1];
 static char *module_search_path = NULL;
 static char lib_python[] = LIB_PYTHON;
+static char share_python[] = "share/python" VERSION;
 
 static void
 reduce(char *dir)
@@ -270,7 +271,7 @@
         delim = strchr(prefix, DELIM);
         if (delim)
             *delim = '\0';
-        joinpath(prefix, lib_python);
+        joinpath(prefix, share_python);
         joinpath(prefix, LANDMARK);
         return 1;
     }
@@ -293,7 +294,7 @@
     copy_absolute(prefix, argv0_path);
     do {
         n = strlen(prefix);
-        joinpath(prefix, lib_python);
+        joinpath(prefix, share_python);
         joinpath(prefix, LANDMARK);
         if (ismodule(prefix))
             return 1;
@@ -303,7 +304,7 @@
 
     /* Look at configure's PREFIX */
     strncpy(prefix, PREFIX, MAXPATHLEN);
-    joinpath(prefix, lib_python);
+    joinpath(prefix, share_python);
     joinpath(prefix, LANDMARK);
     if (ismodule(prefix))
         return 1;
@@ -510,7 +511,7 @@
             fprintf(stderr,
                 "Could not find platform independent libraries <prefix>\n");
         strncpy(prefix, PREFIX, MAXPATHLEN);
-        joinpath(prefix, lib_python);
+        joinpath(prefix, share_python);
     }
     else
         reduce(prefix);
--- Python-2.4.2/Lib/site.py~	2006-01-25 22:39:11.000000000 +0200
+++ Python-2.4.2/Lib/site.py	2006-01-25 23:11:27.000000000 +0200
@@ -182,13 +182,12 @@
                                          sys.lib,
                                          "python" + sys.version[:3],
                                          "site-packages"),
-                            os.path.join(prefix, sys.lib, "site-python")]
-                if sys.lib != 'lib':
-                    sitedirs.append(os.path.join(prefix,
-                                                 'lib',
-                                                 "python" + sys.version[:3],
-                                                 "site-packages"))
-                    sitedirs.append(os.path.join(prefix, 'lib', "site-python"))
+                            os.path.join(prefix, sys.lib, "site-python"),
+                            os.path.join(prefix,
+                                         "share",
+                                         "python" + sys.version[:3],
+                                         "site-packages"),
+                            os.path.join(prefix, "share", "site-python")]
             else:
                 sitedirs = [prefix, os.path.join(prefix, sys.lib, "site-packages")]
             if sys.platform == 'darwin':
--- Python-2.4.2/Modules/getpath.c~	2006-01-26 01:43:31.000000000 +0200
+++ Python-2.4.2/Modules/getpath.c	2006-01-26 01:43:34.000000000 +0200
@@ -113,15 +113,15 @@
 
 #ifndef LIB_PYTHON
 #if defined(__x86_64__)
-#define LIB_PYTHON "lib64/python" VERSION
+#define LIB_PYTHON "lib64/python"
 #else
-#define LIB_PYTHON "lib/python" VERSION
+#define LIB_PYTHON "lib/python"
 #endif
 #endif
 
 #ifndef PYTHONPATH
-#define PYTHONPATH PREFIX "/" LIB_PYTHON ":" \
-              EXEC_PREFIX "/" LIB_PYTHON "/lib-dynload:" \
+#define PYTHONPATH PREFIX "/" LIB_PYTHON VERSION ":" \
+              EXEC_PREFIX "/" LIB_PYTHON VERSION "/lib-dynload:" \
               PREFIX "/share/python" VERSION
 #endif
 
@@ -133,7 +133,7 @@
 static char exec_prefix[MAXPATHLEN+1];
 static char progpath[MAXPATHLEN+1];
 static char *module_search_path = NULL;
-static char lib_python[] = LIB_PYTHON;
+static char lib_python[] = LIB_PYTHON VERSION;
 static char share_python[] = "share/python" VERSION;
 
 static void
@@ -525,7 +525,7 @@
     }
     else
         strncpy(zip_path, PREFIX, MAXPATHLEN);
-    joinpath(zip_path, "lib/python00.zip");
+    joinpath(zip_path, LIB_PYTHON "00.zip");
     bufsz = strlen(zip_path);	/* Replace "00" with version */
     zip_path[bufsz - 6] = VERSION[0];
     zip_path[bufsz - 5] = VERSION[2];
