Index: Lib/socket.py
===================================================================
--- Lib/socket.py	(revision 54542)
+++ Lib/socket.py	(working copy)
@@ -145,6 +145,10 @@
     send = recv = recv_into = sendto = recvfrom = recvfrom_into = _dummy
     __getattr__ = _dummy
 
+# Wrapper around platform socket objects. This implements
+# a platform-independent dup() functionality. The 
+# implementation currently relies on reference counting
+# to close the underlying socket object.
 class _socketobject(object):
 
     __doc__ = _realsocket.__doc__
Index: Lib/httplib.py
===================================================================
--- Lib/httplib.py	(revision 54542)
+++ Lib/httplib.py	(working copy)
@@ -1133,6 +1133,9 @@
     def __getattr__(self, attr):
         return getattr(self._sock, attr)
 
+    def close(self):
+        SharedSocketClient.close(self)
+        self._ssl = None
 
 class HTTPSConnection(HTTPConnection):
     "This class allows communication via SSL."

--- python/trunk/Lib/httplib.py	2006/05/03 18:03:22	45890
+++ python/trunk/Lib/httplib.py	2006/07/26 12:12:56	50844
@@ -926,8 +926,8 @@
         self.__state = _CS_IDLE
 
         if response.will_close:
-            # this effectively passes the connection to the response
-            self.close()
+            # Pass the socket to the response
+            self.sock = None
         else:
             # remember this, so we can tell when it is complete
             self.__response = response
