--- Python-1.5.2/Lib/cgi.py.pythonpath	Fri Jan  8 12:42:03 1999
+++ Python-1.5.2/Lib/cgi.py	Mon Aug  9 23:27:37 1999
@@ -1,4 +1,4 @@
-#! /usr/local/bin/python
+#! /usr/bin/python
 
 """Support module for CGI (Common Gateway Interface) scripts.
 
--- Python-1.5.2/Misc/HISTORY.pythonpath	Mon Dec 21 16:41:14 1998
+++ Python-1.5.2/Misc/HISTORY	Mon Aug  9 23:27:40 1999
@@ -1333,9 +1333,9 @@
 were invoked.
 
 - It is now recommended to use ``#!/usr/bin/env python'' instead of
-``#!/usr/local/bin/python'' at the start of executable scripts, except
+``#!/usr/bin/python'' at the start of executable scripts, except
 for CGI scripts.  It has been determined that the use of /usr/bin/env
-is more portable than that of /usr/local/bin/python -- scripts almost
+is more portable than that of /usr/bin/python -- scripts almost
 never have to be edited when the Python interpreter lives in a
 non-standard place.  Note that this doesn't work for CGI scripts since
 the python executable often doesn't live in the HTTP server's default
@@ -4619,7 +4619,7 @@
 Python exits or receives a SIGTERM or SIGHUP signal.
 
 The interpreter is now generally assumed to live in
-/usr/local/bin/python (as opposed to /usr/local/python).  The script
+/usr/bin/python (as opposed to /usr/local/python).  The script
 demo/scripts/fixps.py will update old scripts in place (you can easily
 modify it to do other similar changes).
 
--- Python-1.5.2/Tools/faqwiz/faqw.py.pythonpath	Fri Apr  3 17:27:04 1998
+++ Python-1.5.2/Tools/faqwiz/faqw.py	Mon Aug  9 23:27:42 1999
@@ -1,4 +1,4 @@
-#! /usr/local/bin/python
+#! /usr/bin/python
 
 """FAQ wizard bootstrap."""
 
@@ -10,7 +10,7 @@
 # executable.
 
 # You need to edit the first line and the lines that define FAQDIR and
-# SRCDIR, below: change /usr/local/bin/python to where your Python
+# SRCDIR, below: change /usr/bin/python to where your Python
 # interpreter lives, change the value for FAQDIR to where your FAQ
 # lives, and change the value for SRCDIR to where your faqwiz.py
 # module lives.  The faqconf.py and faqcust.py files live there, too.
--- Python-1.5.2/Tools/scripts/README.pythonpath	Wed Oct  7 15:52:35 1998
+++ Python-1.5.2/Tools/scripts/README	Mon Aug  9 23:27:43 1999
@@ -30,7 +30,7 @@
 ndiff.py		Intelligent diff between text files (Tim Peters)
 nm2def.py		Create a template for PC/python_nt.def (Marc Lemburg)
 objgraph.py		Print object graph from nm output on a library
-pathfix.py		Change #!/usr/local/bin/python into something else
+pathfix.py		Change #!/usr/bin/python into something else
 pdeps.py		Print dependencies between Python modules
 pindent.py		Indent Python code, giving block-closing comments
 ptags.py		Create vi tags file for Python modules
--- Python-2.1/Tools/scripts/fixps.py.pythonpath	Tue Apr 24 10:22:56 2001
+++ Python-2.1/Tools/scripts/fixps.py	Tue Apr 24 10:24:43 2001
@@ -15,13 +15,13 @@
             print file, ': can\'t open :', msg
             continue
         line = f.readline()
-        if not re.match('^#! */usr/local/bin/python', line):
-            print file, ': not a /usr/local/bin/python script'
+        if not re.match('^#! */usr/bin/python', line):
+            print file, ': not a /usr/bin/python script'
             f.close()
             continue
         rest = f.read()
         f.close()
-        line = re.sub('/usr/local/bin/python',
+        line = re.sub('/usr/bin/python',
                       '/usr/bin/env python', line)
         print file, ':', `line`
         f = open(file, "w")
