--- Python-2.4.2/setup.py.orig	2005-11-22 14:02:49.000000000 +0000
+++ Python-2.4.2/setup.py	2005-11-22 16:06:42.060536936 +0000
@@ -489,7 +489,7 @@
         #
         # http://www.sleepycat.com/update/index.html
 
-        max_db_ver = (4, 3)
+        max_db_ver = (4, 4)
         min_db_ver = (3, 2)
         db_setup_debug = False   # verbose debug prints from this script?
 
@@ -506,7 +506,7 @@
             '/sw/include/db3',
         ]
         # 4.x minor number specific paths
-        for x in (0,1,2,3):
+        for x in (0,1,2,3,4):
             db_inc_paths.append('/usr/include/db4%d' % x)
             db_inc_paths.append('/usr/local/BerkeleyDB.4.%d/include' % x)
             db_inc_paths.append('/usr/local/include/db4%d' % x)
--- Python-2.4.2/Modules/_bsddb.c.orig	2004-12-16 09:48:37.000000000 +0000
+++ Python-2.4.2/Modules/_bsddb.c	2005-11-22 16:39:39.734884736 +0000
@@ -97,7 +97,7 @@
 #error "eek! DBVER can't handle minor versions > 9"
 #endif
 
-#define PY_BSDDB_VERSION "4.3.0"
+#define PY_BSDDB_VERSION "4.4.0"
 static char *rcs_id = "$Id$";
 
 
@@ -4101,8 +4101,15 @@
 #endif
     MAKE_ENTRY(nrequests);
     MAKE_ENTRY(nreleases);
+#if (DBVER < 44)
     MAKE_ENTRY(nnowaits);
     MAKE_ENTRY(nconflicts);
+#else
+    MAKE_ENTRY(nupgrade);
+    MAKE_ENTRY(ndowngrade);
+    MAKE_ENTRY(lock_wait);
+    MAKE_ENTRY(lock_nowait);
+#endif
     MAKE_ENTRY(ndeadlocks);
     MAKE_ENTRY(regsize);
     MAKE_ENTRY(region_wait);
@@ -4938,7 +4945,13 @@
     ADD_INT(d, DB_LOCK_IREAD);
     ADD_INT(d, DB_LOCK_IWR);
 #if (DBVER >= 33)
+#if (DBVER < 44)
     ADD_INT(d, DB_LOCK_DIRTY);
+#else
+    ADD_INT(d, DB_LOCK_READ_UNCOMMITTED);
+#endif
+#endif
+#if (DBVER >= 33)
     ADD_INT(d, DB_LOCK_WWRITE);
 #endif
 
