diff -Nur Python-2.5.orig/Lib/distutils/command/install.py Python-2.5/Lib/distutils/command/install.py
--- Python-2.5.orig/Lib/distutils/command/install.py	2007-03-17 16:26:13.803357750 +0000
+++ Python-2.5/Lib/distutils/command/install.py	2007-03-17 16:26:50.821671250 +0000
@@ -40,14 +40,14 @@
 
 INSTALL_SCHEMES = {
     'unix_prefix': {
-        'purelib': '$base/lib/python$py_version_short/site-packages',
+        'purelib': '$base/share/python$py_version_short/site-packages',
         'platlib': '$platbase/'+libname+'/python$py_version_short/site-packages',
         'headers': '$base/include/python$py_version_short/$dist_name',
         'scripts': '$base/bin',
         'data'   : '$base',
         },
     'unix_home': {
-        'purelib': '$base/lib/python',
+        'purelib': '$base/share/python',
         'platlib': '$base/'+libname+'/python',
         'headers': '$base/include/python/$dist_name',
         'scripts': '$base/bin',
diff -Nur Python-2.5.orig/Lib/distutils/sysconfig.py Python-2.5/Lib/distutils/sysconfig.py
--- Python-2.5.orig/Lib/distutils/sysconfig.py	2007-03-17 16:26:13.803357750 +0000
+++ Python-2.5/Lib/distutils/sysconfig.py	2007-03-17 16:26:50.821671250 +0000
@@ -99,12 +99,12 @@
         prefix = plat_specific and EXEC_PREFIX or PREFIX
 
     if os.name == "posix":
-        if plat_specific:
-            lib = sys.lib
+        if plat_specific: 
+            libpython = os.path.join(prefix,
+                                 sys.lib, "python" + get_python_version())
         else:
-            lib = 'lib'
-        libpython = os.path.join(prefix,
-                                 lib, "python" + get_python_version())
+            libpython = os.path.join(prefix,
+                                 "share", "python" + get_python_version())
         if standard_lib:
             return libpython
         else:
diff -Nur Python-2.5.orig/Lib/site.py Python-2.5/Lib/site.py
--- Python-2.5.orig/Lib/site.py	2007-03-17 16:26:13.803357750 +0000
+++ Python-2.5/Lib/site.py	2007-03-17 16:26:50.825671500 +0000
@@ -269,13 +269,10 @@
                                         "python" + sys.version[:3],
                                         "site-packages"))
             sitepackages.append(os.path.join(prefix, sys.lib, "site-python"))
-            if sys.lib != 'lib':
-                sitepackages.append(os.path.join(prefix,
-                    'lib',
-                    "python" + sys.version[:3],
-                    "site-packages"))
-                sitepackages.append(os.path.join(prefix, 'lib', "site-python"))
-
+            sitepackages.append(os.path.join(prefix, "share",
+                "python" + sys.version[:3],
+                "site-packages"))
+            sitepackages.append(os.path.join(prefix, "share", "site-python"))
         else:
             sitepackages.append(prefix)
             sitepackages.append(os.path.join(prefix, sys.lib, "site-packages"))
--- p/Lib/test/test_site.py~	2012-05-10 18:02:09.000000000 +0200
+++ p/Lib/test/test_site.py	2012-05-10 18:09:38.776569290 +0200
@@ -241,7 +241,7 @@
             self.assertEqual(dirs[2], wanted)
         elif os.sep == '/':
             # OS X non-framwework builds, Linux, FreeBSD, etc
-            self.assertEqual(len(dirs), 2)
+            self.assertEqual(len(dirs), 4)
             wanted = os.path.join('xoxo', sys.lib, 'python' + sys.version[:3],
                                   'site-packages')
             self.assertEqual(dirs[0], wanted)
