--- Python-2.2.3/setup.py.orig	Thu May 22 19:36:54 2003
+++ Python-2.2.3/setup.py	Sun Jun  1 22:00:30 2003
@@ -400,17 +400,7 @@
 
         # The standard Unix dbm module:
         if platform not in ['cygwin']:
-            if (self.compiler.find_library_file(lib_dirs, 'ndbm')):
-                exts.append( Extension('dbm', ['dbmmodule.c'],
-                                       libraries = ['ndbm'] ) )
-            elif self.compiler.find_library_file(lib_dirs, 'db1'):
-                exts.append( Extension('dbm', ['dbmmodule.c'],
-                                       libraries = ['db1'] ) )
-            elif self.compiler.find_library_file(lib_dirs, 'gdbm'):
-                exts.append( Extension('dbm', ['dbmmodule.c'],
-                                       libraries = ['gdbm'] ) )
-            else:
-                exts.append( Extension('dbm', ['dbmmodule.c']) )
+            exts.append( Extension('dbm', ['dbmmodule.c']) )
 
         # Anthony Baxter's gdbm module.  GNU dbm(3) will require -lgdbm:
         if (self.compiler.find_library_file(lib_dirs, 'gdbm')):
