--- Python-2.0b2/Modules/Setup.config.in.wiget	Fri Sep  8 04:17:15 2000
+++ Python-2.0b2/Modules/Setup.config.in	Tue Oct 10 15:42:50 2000
@@ -12,7 +12,7 @@
 # You may want this to be built as a dynamically loaded module; uncomment
 # the following line in that case:
 
-#*shared*
+*shared*
 
 # bsddb module enabled by --with-libdb or presence of db.h
 @USE_BSDDB_MODULE@bsddb bsddbmodule.c @HAVE_LIBDB@
