--- Python-2.4a3/Lib/site.py~	2004-07-20 02:28:28.000000000 +0000
+++ Python-2.4a3/Lib/site.py	2004-09-22 10:58:42.845270232 +0000
@@ -350,7 +350,7 @@
     default is 'ascii', but if you're willing to experiment, you can
     change this."""
     encoding = "ascii" # Default value set by _PyUnicode_Init()
-    if 0:
+    if 1:
         # Enable to support locale aware default string encodings.
         import locale
         loc = locale.getdefaultlocale()
