diff -Nur Python-2.1.orig/Makefile.pre.in Python-2.1/Makefile.pre.in
--- Python-2.1.orig/Makefile.pre.in	Fri May 25 10:48:51 2001
+++ Python-2.1/Makefile.pre.in	Fri May 25 10:49:48 2001
@@ -745,7 +745,7 @@
 # This goes into $(exec_prefix)
 sharedinstall:
 	PYTHONPATH= $(RUNSHARED) ./$(PYTHON) $(srcdir)/setup.py install \
-		--install-platlib=$(DESTSHARED)
+		--install-platlib=$(DESTSHARED) --install-scripts=$(BINDIR)
 
 # Build the toplevel Makefile
 Makefile.pre: Makefile.pre.in config.status
