diff -urN Python-2.3.1.org/configure.in Python-2.3.1/configure.in
--- Python-2.3.1.org/configure.in	2003-09-27 12:45:29.550715080 +0200
+++ Python-2.3.1/configure.in	2003-09-27 12:45:53.014148096 +0200
@@ -2093,7 +2093,7 @@
 )
 AC_MSG_CHECKING(for fsync)
 AC_TRY_COMPILE([#include <unistd.h>], void *x=fsync,
-  AC_DEFINE(HAVE_SYNC, 1, Define if you have the 'fsync' function.)
+  AC_DEFINE(HAVE_FSYNC, 1, Define if you have the 'fsync' function.)
   AC_MSG_RESULT(yes),
   AC_MSG_RESULT(no)
 )
diff -urN Python-2.3.1.org/pyconfig.h.in Python-2.3.1/pyconfig.h.in
--- Python-2.3.1.org/pyconfig.h.in	2003-09-27 12:45:29.623703984 +0200
+++ Python-2.3.1/pyconfig.h.in	2003-09-27 12:46:09.172691624 +0200
@@ -486,7 +486,7 @@
 #undef HAVE_SYMLINK
 
 /* Define if you have the 'fsync' function. */
-#undef HAVE_SYNC
+#undef HAVE_FSYNC
 
 /* Define to 1 if you have the `sysconf' function. */
 #undef HAVE_SYSCONF
