--- Python-2.3.3/Python/sysmodule.c.orig	2003-08-09 11:48:29.000000000 +0200
+++ Python-2.3.3/Python/sysmodule.c	2004-05-02 13:54:01.000000000 +0200
@@ -1209,12 +1209,6 @@
 #endif /* Unix */
 		}
 #endif /* All others */
-		a = PyString_FromStringAndSize(argv0, n);
-		if (a == NULL)
-			Py_FatalError("no mem for sys.path insertion");
-		if (PyList_Insert(path, 0, a) < 0)
-			Py_FatalError("sys.path.insert(0) failed");
-		Py_DECREF(a);
 	}
 	Py_DECREF(av);
 }
