diff -dur Python-2.3.4.orig/Lib/distutils/command/install.py Python-2.3.4/Lib/distutils/command/install.py
--- Python-2.3.4.orig/Lib/distutils/command/install.py	2004-09-10 08:50:27.601692800 +0200
+++ Python-2.3.4/Lib/distutils/command/install.py	2004-09-10 08:51:00.726881742 +0200
@@ -39,14 +39,14 @@
 INSTALL_SCHEMES = {
     'unix_prefix': {
         'purelib': '$base/share/python$py_version_short/site-packages',
-        'platlib': '$platbase/lib/python$py_version_short/site-packages',
+        'platlib': '$platbase/lib64/python$py_version_short/site-packages',
         'headers': '$base/include/python$py_version_short/$dist_name',
         'scripts': '$base/bin',
         'data'   : '$base',
         },
     'unix_home': {
         'purelib': '$base/share/python',
-        'platlib': '$base/lib/python',
+        'platlib': '$base/lib64/python',
         'headers': '$base/include/python/$dist_name',
         'scripts': '$base/bin',
         'data'   : '$base',
diff -dur Python-2.3.4.orig/Lib/distutils/sysconfig.py Python-2.3.4/Lib/distutils/sysconfig.py
--- Python-2.3.4.orig/Lib/distutils/sysconfig.py	2004-09-10 08:50:27.602692745 +0200
+++ Python-2.3.4/Lib/distutils/sysconfig.py	2004-09-10 08:51:46.443382256 +0200
@@ -101,7 +101,7 @@
     if os.name == "posix":
 	if plat_specific: 
             libpython = os.path.join(prefix,
-                                 "lib", "python" + get_python_version())
+                                 "lib64", "python" + get_python_version())
        else:
             libpython = os.path.join(prefix,
                                  "share", "python" + get_python_version())
diff -dur Python-2.3.4.orig/Lib/site.py Python-2.3.4/Lib/site.py
--- Python-2.3.4.orig/Lib/site.py	2004-09-10 08:50:27.603692691 +0200
+++ Python-2.3.4/Lib/site.py	2004-09-10 08:51:00.729881578 +0200
@@ -166,10 +166,10 @@
             sitedirs = [os.path.join(prefix, "Lib", "site-packages")]
         elif os.sep == '/':
             sitedirs = [os.path.join(prefix,
-                                     "lib",
+                                     "lib64",
                                      "python" + sys.version[:3],
                                      "site-packages"),
-                        os.path.join(prefix, "lib", "site-python"),
+                        os.path.join(prefix, "lib64", "site-python"),
                         os.path.join(prefix,
                                      "share",
                                      "python" + sys.version[:3],
diff -dur Python-2.3.4.orig/Makefile.pre.in Python-2.3.4/Makefile.pre.in
--- Python-2.3.4.orig/Makefile.pre.in	2004-09-10 08:50:27.604692636 +0200
+++ Python-2.3.4/Makefile.pre.in	2004-09-10 08:51:00.730881524 +0200
@@ -79,7 +79,7 @@
 
 # Expanded directories
 BINDIR=		$(exec_prefix)/bin
-LIBDIR=		$(exec_prefix)/lib
+LIBDIR=		$(exec_prefix)/lib64
 MANDIR=		@mandir@
 INCLUDEDIR=	@includedir@
 CONFINCLUDEDIR=	$(exec_prefix)/include
diff -dur Python-2.3.4.orig/Modules/getpath.c Python-2.3.4/Modules/getpath.c
--- Python-2.3.4.orig/Modules/getpath.c	2004-09-10 08:50:27.605692581 +0200
+++ Python-2.3.4/Modules/getpath.c	2004-09-10 08:51:00.731881469 +0200
@@ -113,7 +113,7 @@
 
 #ifndef PYTHONPATH
 #define PYTHONPATH PREFIX "/share/python" VERSION ":" \
-              EXEC_PREFIX "/lib/python" VERSION "/lib-dynload"
+              EXEC_PREFIX "/lib64/python" VERSION "/lib-dynload"
 #endif
 
 #ifndef LANDMARK
@@ -124,7 +124,7 @@
 static char exec_prefix[MAXPATHLEN+1];
 static char progpath[MAXPATHLEN+1];
 static char *module_search_path = NULL;
-static char lib_python[] = "lib/python" VERSION;
+static char lib_python[] = "lib64/python" VERSION;
 static char share_python[] = "share/python" VERSION;
 
 static void
@@ -493,7 +493,7 @@
     }
     else
         strncpy(zip_path, PREFIX, MAXPATHLEN);
-    joinpath(zip_path, "lib/python00.zip");
+    joinpath(zip_path, "lib64/python00.zip");
     bufsz = strlen(zip_path);	/* Replace "00" with version */
     zip_path[bufsz - 6] = VERSION[0];
     zip_path[bufsz - 5] = VERSION[2];
Only in Python-2.3.4.orig: Python-Docs-2.3.4
diff -dur Python-2.3.4.orig/setup.py Python-2.3.4/setup.py
--- Python-2.3.4.orig/setup.py	2004-09-10 08:50:27.562694932 +0200
+++ Python-2.3.4/setup.py	2004-09-10 08:51:00.733881359 +0200
@@ -264,7 +264,7 @@
         # lib_dirs and inc_dirs are used to search for files;
         # if a file is found in one of those directories, it can
         # be assumed that no additional -I,-L directives are needed.
-        lib_dirs = self.compiler.library_dirs + ['/lib', '/usr/lib']
+        lib_dirs = self.compiler.library_dirs + ['/lib64', '/usr/lib64']
         inc_dirs = self.compiler.include_dirs + ['/usr/include']
         exts = []
 
@@ -966,7 +966,7 @@
             added_lib_dirs.append('/usr/openwin/lib')
         elif os.path.exists('/usr/X11R6/include'):
             include_dirs.append('/usr/X11R6/include')
-            added_lib_dirs.append('/usr/X11R6/lib')
+            added_lib_dirs.append('/usr/X11R6/lib64')
         elif os.path.exists('/usr/X11R5/include'):
             include_dirs.append('/usr/X11R5/include')
             added_lib_dirs.append('/usr/X11R5/lib')
