--- Python-2.4.3/Objects/unicodeobject.c.orig	2006-05-13 20:34:03.118998250 +0200
+++ Python-2.4.3/Objects/unicodeobject.c	2006-05-13 20:33:52.970364000 +0200
@@ -2302,7 +2302,7 @@
     end = s + size;
 
     while (s < end) {
-        *p = *(Py_UNICODE *)s;
+        memcpy(p,s,sizeof(Py_UNICODE));
         /* We have to sanity check the raw data, otherwise doom looms for
            some malformed UCS-4 data. */
         if (
