diff -Nur Python-1.6b1.orig/Python/dynload_shlib.c Python-1.6b1/Python/dynload_shlib.c
--- Python-1.6b1.orig/Python/dynload_shlib.c	Thu Aug  3 18:50:35 2000
+++ Python-1.6b1/Python/dynload_shlib.c	Tue Aug  8 14:08:21 2000
@@ -72,12 +72,12 @@
 #ifdef RTLD_NOW
 	/* RTLD_NOW: resolve externals now
 	   (i.e. core dump now if some are missing) */
-	handle = dlopen(pathname, RTLD_NOW);
+	handle = dlopen(pathname, RTLD_NOW | RTLD_GLOBAL);
 #else
 	if (Py_VerboseFlag)
 		printf("dlopen(\"%s\", %d);\n", pathname,
-		       RTLD_LAZY);
-	handle = dlopen(pathname, RTLD_LAZY);
+		       RTLD_LAZY | RTLD_GLOBAL);
+	handle = dlopen(pathname, RTLD_LAZY | RTLD_GLOBAL);
 #endif /* RTLD_NOW */
 	if (handle == NULL) {
 		PyErr_SetString(PyExc_ImportError, dlerror());
